var/spack/repos/builtin/packages/tiramisu/package.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import llnl.util.filesystem as fs
+from spack.package import *
+
+
+class Tiramisu(CMakePackage, PythonExtension):
+    """Tiramisu is a polyhedral compiler for dense and sparse deep learning
+    and data parallel algorithms.It provides a simple C++ API for expressing
+    algorithms and how these algorithms should be optimized by the compiler."""
+
+    homepage = "http://tiramisu-compiler.org"
+    url = "https://github.com/Tiramisu-Compiler/tiramisu"
+    git = "https://github.com/Tiramisu-Compiler/tiramisu.git"
+
+    maintainers = ["wraith1995"]
+
+    generator = "Ninja"
+
+    version("master", branch="master")
+    version("2023-2-3", commit="73a9cec72e08d4dfe5e8c66da33139008124a4fa")
+
+    variant("python", default=True, description="Install python bindings.")
+    variant("cuda", default=False, description="Enable Cuda Code Generation.")
+    extends("python", when="+python")
+    variant(
+        "debug",
+        default="0",
+        description="Set the debug level.",
+        multi=False,
+        values=("0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"),
+    )
+
+    depends_on("cmake@3.24:", type="build")
+    depends_on("ninja", type="build")
+
+    depends_on("halide@14.0.0", type=("build", "link", "run"))
+    depends_on("isl", type=("build", "link", "run"))
+    depends_on("cuda", when="+cuda", type=("build", "link", "run"))
+
+    depends_on("python@3.8:", type=("build", "link", "run"), when="+python")
+    depends_on("py-pybind11@2.6.2:", type=("build", "link", "run"), when="+python")
+    depends_on("py-numpy", type=("build", "run"), when="+python")
+    depends_on("py-cython", type=("run"), when="+python")
```

Comment:
pybind11 and cython are rarely needed at run-time. Where did find the full list of deps?

Reply:
Pybind is indeed only build time, but Cython is actually used to build python extensions for the code that Tiramisu generates. 

In this case, the full list of depends is only partially detailed in the readme at the moment. I fix the readme if so desired.

Reply:
I am making a pull to update the readme with the current dependencies.